src/parameters/tileparameter_nb.cpp

```diff
@@ -0,0 +1,28 @@
+#include "parameters/tileparameter_nb.h"
+
+#include "engine/api/tile_parameters.hpp"
+
+#include <nanobind/nanobind.h>
+#include <nanobind/stl/string.h>
+#include <nanobind/stl/vector.h>
+
+namespace nb = nanobind;
+using namespace nb::literals;
+
+void init_TileParameters(nb::module_& m) {
+    using osrm::engine::api::TileParameters;
+
+    nb::class_<TileParameters>(m, "TileParameters", nb::is_final())
+        .def(nb::init<>())
+        .def(nb::init<unsigned, unsigned, unsigned>(),
+            "x"_a, "y"_a, "z"_a
+        )
+        .def("__init__", [](TileParameters* t, const std::vector<unsigned>& coord) {
+            new (t) TileParameters{coord[0], coord[1], coord[2]};
+         })
```

Comment:
just for reference: better to check in the function body if it was exactly 3 parameters passed, no more/less. also the result here is very likely binary protobuf

Reply:
Okay, I've adjusted this to validate the vector, and throw a runtime error if the size is not 3.